tests/e2e/functional_test.go

```diff
@@ -289,6 +329,14 @@ var _ = Describe("Functional e2e", Serial, func() {
 			return rollout, nil
 		})
 
+		if dataLossPrevention == "true" {
+			document("Verify that Pipelines are being paused by checking rollout condition")
+			Eventually(func() metav1.ConditionStatus {
+				rollout, _ := numaflowControllerRolloutClient.Get(ctx, numaflowControllerRolloutName, metav1.GetOptions{})
+				return getRolloutCondition(rollout.Status.Conditions, apiv1.ConditionPausingPipelines)
+			}, testTimeout, testPollingInterval).Should(Equal(metav1.ConditionTrue))
+		}
```

Comment:
do you want to confirm that this goes to "False" later? 

Also, in the same `Eventually` statement here, you could also check that the Pipeline's Condition is True here

Reply:
We confirm its false later by calling `verifyNumaflowControllerReady` on L346.

For checking the Pipeline's condition, do you want to check the rollout or the child Pipeline?

Reply:
sorry, I just saw you did check that it goes to "False" later

Reply:
```
if dataLossPrevention == "true" {
			document("Verify that Pipelines are being paused by checking rollout condition")
			Eventually(func() bool {
				isbRollout, _ := isbServiceRolloutClient.Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
				isbStatus := getRolloutCondition(isbRollout.Status.Conditions, apiv1.ConditionPausingPipelines)
				plRollout, _ := pipelineRolloutClient.Get(ctx, pipelineRolloutName, metav1.GetOptions{})
				plStatus := getRolloutCondition(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
				if isbStatus != metav1.ConditionTrue || plStatus != metav1.ConditionTrue {
					return false
				}
				return true
			}, testTimeout).Should(BeTrue())
		}
```

Reply:
Just realized there is also the `verifyPipelineStatus` func that would probably be better for this. My only hesitation is that there is a call to document every time that function is run so we might be printing the state of the cluster many many times if this is called quite often. I could move that call out of that function and into `verifyPipelineReady` where it's being called.

Thoughts? @juliev0

Reply:
> For checking the Pipeline's condition, do you want to check the rollout or the child Pipeline?
Hmm, I'm okay either way.

Reply:
> Just realized there is also the verifyPipelineStatus func that would probably be better for this. My only hesitation is that there is a call to document every time that function is run so we might be printing the state of the cluster many many times if this is called quite often. I could move that call out of that function and into verifyPipelineReady where it's being called.

Good calls. Yeah, we could move the `document` call out of `verifyPipelineStatus()`

Reply:
This will mean we would wrap an Eventually inside another. Maybe it would be best we just check the status of the Pipeline before or after we check the rollout condition.